pkg/imageserver/webdav_test.go

```diff
@@ -0,0 +1,234 @@
+package imageserver
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	docker "github.com/fsouza/go-dockerclient"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/image-inspector/pkg/api"
+)
+
+const (
+	versionTag             = "v1"
+	healthzPath            = "/healthz"
+	apiPrefix              = "/api"
+	contentPath            = apiPrefix + "/" + versionTag + "/content/"
+	metadataPath           = apiPrefix + "/" + versionTag + "/metadata"
+	openscapReportPath     = apiPrefix + "/" + versionTag + "/openscap"
+	openScapHTMLReportPath = apiPrefix + "/" + versionTag + "/openscap-report"
+	scanType               = "openscap"
+	authToken              = "12345"
+)
+
+var _ = Describe("Webdav", func() {
+	var (
+		server           *httptest.Server
+		options          ImageServerOptions
+		dstPath          string
+		dummyScanResults = api.ScanResult{
+			APIVersion: api.DefaultResultsAPIVersion,
+			Results:    []api.Result{},
+		}
+		dummyMetadata = &api.InspectorMetadata{
+			Image: docker.Image{
+				ID: "dummy",
+			},
+			OpenSCAP: &api.OpenSCAPMetadata{
+				Status: api.StatusSuccess,
+			},
+		}
+		dummyScanReport     = []byte("this is a dummy scan report")
+		dummyHTMLScanReport = []byte("this is a dummy HTML scan report")
+		apiVersions         = api.APIVersions{Versions: []string{versionTag}}
+	)
+	JustBeforeEach(func() {
+		var err error
+		dstPath, err = ioutil.TempDir("", "")
+		Expect(err).NotTo(HaveOccurred())
+		options = ImageServerOptions{
+			HealthzURL:        healthzPath,
+			APIURL:            apiPrefix,
+			APIVersions:       apiVersions,
+			MetadataURL:       metadataPath,
+			ContentURL:        contentPath,
+			ScanType:          scanType,
+			ScanReportURL:     openscapReportPath,
+			HTMLScanReport:    true,
+			HTMLScanReportURL: openScapHTMLReportPath,
+			AuthToken:         authToken,
+			Chroot:            false,
+		}
+		handler, err := NewWebdavImageServer(options).(*webdavImageServer).GetHandler(dummyMetadata, dstPath, dummyScanResults, dummyScanReport, dummyHTMLScanReport)
+		Expect(err).NotTo(HaveOccurred())
+		server = httptest.NewServer(handler)
+	})
+	AfterEach(func() {
+		server.Close()
+		os.RemoveAll(dstPath)
+	})
+	Describe("Endpoints:", func() {
+		var u *url.URL
+		var expBody []byte
+		var status int
+		var body []byte
+		var err error
```

Comment:
any reason you wanted to reuse these items?  I'd prefer we just avoid any shadowing that may occur and cause issues down the road.